services/getstream/followTopic.js

```diff
@@ -51,22 +51,41 @@ const addTopicToChatTab = async (token, topicName, userId, isAnonymous) => {
       : 'This topic has new followers';
     const defaultImage = isAnonymous ? DEFAULT_TOPIC_PIC_PATH_ANON : DEFAULT_TOPIC_PIC_PATH_SIGN;
 
+    const admin = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
+    const adminChannel = admin.channel('topics', `topic_${topic}`);
+    const adminChannelData = await adminChannel.create();
+
+    const updateData = {};
+    // Update channel image if null to default image
+    if (!adminChannelData?.channel?.image) updateData.image = defaultImage;
+    if (!adminChannelData?.channel?.channelImage) updateData.channelImage = defaultImage;
+    if (!adminChannelData?.channel?.channel_image) updateData.channel_image = defaultImage;
+
+    // Revert back all existing signed channel that has anon topic to correct signed one
+    if (!isAnonymous && adminChannelData?.channel?.image === DEFAULT_TOPIC_PIC_PATH_ANON) {
+      updateData.image = DEFAULT_TOPIC_PIC_PATH_SIGN;
+      updateData.channelImage = DEFAULT_TOPIC_PIC_PATH_SIGN;
+      updateData.channel_image = DEFAULT_TOPIC_PIC_PATH_SIGN;
+    }
+
+    await adminChannel.updatePartial({
+      set: updateData
+    });
+
     const client = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
     await client.connectUser({id: userId}, token);
     const channel = await client.channel('topics', `topic_${topic}`, {
       name: `#${topic}`,
       members: [userId],
-      channel_type: 3,
-      channel_image: defaultImage,
-      channelImage: defaultImage,
-      image: defaultImage
+      channel_type: 3
     });
 
     console.log('prepare follow channel', topic);
     await channel.create();
     await channel.addMembers([userId]);
     await channel.sendMessage({text}, {skip_push: true});
     console.log('channel followed');
+    await client.disconnectUser();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code creates two instances of `StreamChat` (one as `admin` and another as `client`). This seems unnecessary and could potentially lead to performance issues. You can use a single instance to perform all the operations. Also, error handling is missing for the async operations which might cause unhandled promise rejections.

```diff
-    const admin = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
-    const adminChannel = admin.channel('topics', `topic_${topic}`);
-    const adminChannelData = await adminChannel.create();
+    const client = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
+    await client.connectUser({id: userId}, token);
+    const channel = client.channel('topics', `topic_${topic}`);
+    const channelData = await channel.create();

-    const updateData = {};
-    // Update channel image if null to default image
-    if (!adminChannelData?.channel?.image) updateData.image = defaultImage;
-    if (!adminChannelData?.channel?.channelImage) updateData.channelImage = defaultImage;
-    if (!adminChannelData?.channel?.channel_image) updateData.channel_image = defaultImage;

-    // Revert back all existing signed channel that has anon topic to correct signed one
-    if (!isAnonymous && adminChannelData?.channel?.image === DEFAULT_TOPIC_PIC_PATH_ANON) {
-      updateData.image = DEFAULT_TOPIC_PIC_PATH_SIGN;
-      updateData.channelImage = DEFAULT_TOPIC_PIC_PATH_SIGN;
-      updateData.channel_image = DEFAULT_TOPIC_PIC_PATH_SIGN;
-    }

-    await adminChannel.updatePartial({
-      set: updateData
-    });

-    const client = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
-    await client.connectUser({id: userId}, token);
-    const channel = await client.channel('topics', `topic_${topic}`, {
+    const updateData = {};
+    // Update channel image if null to default image
+    if (!channelData?.channel?.image) updateData.image = defaultImage;
+    if (!channelData?.channel?.channelImage) updateData.channelImage = defaultImage;
+    if (!channelData?.channel?.channel_image) updateData.channel_image = defaultImage;

+    // Revert back all existing signed channel that has anon topic to correct signed one
+    if (!isAnonymous && channelData?.channel?.image === DEFAULT_TOPIC_PIC_PATH_ANON) {
+      updateData.image = DEFAULT_TOPIC_PIC_PATH_SIGN;
+      updateData.channelImage = DEFAULT_TOPIC_PIC_PATH_SIGN;
+      updateData.channel_image = DEFAULT_TOPIC_PIC_PATH_SIGN;
+    }

+    try {
+      await channel.updatePartial({
+        set: updateData
+      });
+    } catch (error) {
+      console.log('error updating channel data', error);
+      throw new Error('error updating channel data');
+    }

     const channel = client.channel('topics', `topic_${topic}`, {
       name: `#${topic}`,
       members: [userId],
       channel_type: 3
     });

     console.log('prepare follow channel', topic);
     try {
       await channel.create();
       await channel.addMembers([userId]);
       await channel.sendMessage({text}, {skip_push: true});
     } catch (error) {
       console.log('error creating and sending message to channel', error);
       throw new Error('error creating and sending message to channel');
     }
     console.log('channel followed');
     await client.disconnectUser();
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->